File: C_Programming/Practise_Programs/PP_150.c
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_150
//
//  Description       : This program uses a hexadecimal bit mask to
//                      check whether the 17th bit of a given unsigned
//                      integer is ON or OFF.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 18/08/2026
//
//  Time Complexity   : O(1)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

#include<stdio.h> 

typedef unsigned int UINT;

int main()
{
    UINT iNo = 0;               
    UINT iMask = 0x00010000;    
    UINT iAns = 0;

    printf("Enter Number : \n");
    scanf("%d", &iNo);

    iAns = iNo & iMask;

    if(iAns == iMask)
    {
        printf("17th bit is ON\n");
    }
    else
    {
        printf("17th bit is OFF\n");
    }

    return 0;
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      65536
//
//  Sample Output :
//
//      Enter Number :
//      17th bit is ON
//
//////////////////////////////////////////////////////////////////

File: Java_Programming/Practise_Programs/PP_150.java
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_150
//
//  Description       : This program prints a star pattern using
//                      nested for loops. The pattern starts with
//                      a single star and increases the number of
//                      stars in each successive row.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 02/09/2026
//
//  Time Complexity   : O(n^2)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

class PP_150
{
    public static void main(String A[])
    {
        int i, j, k = 0;

        for(i = 1; i <= 5; i++)
        {
            for(j = 1; j <= 5 - i; j++)
            {
                System.out.print(" ");
            }
            for(k = 1; k <= i; k++)
            {

                System.out.print("*");
            }
            System.out.println();
        }
    }
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      No input required.
//
//  Sample Output :
//
//          *
//         **
//        ***
//       ****
//      *****
//
//////////////////////////////////////////////////////////////////